src/utility/bart_hasher.h

```diff
@@ -0,0 +1,53 @@
+#ifndef BART_SRC_UTILITY_BART_HASHER_H_
+#define BART_SRC_UTILITY_BART_HASHER_H_
+
+#include <utility>
+#include <tuple>
+#include <vector>
+
+#include <boost/functional/hash.hpp>
+
+//! Provides various utility functions for BART.
+namespace butil {
+
+//! Hashing struct for array<int> of any size, and vector<int>.
+/*! Provides a hashing struct for array<int> of any size, and vector<int>;
+ *  Suitable for use in an unordered_map. Uses the hashing functions of the boost
+ *  library.
+ *  Example:
+ *  ~~~~~~~~~~{.cpp}
+ *  #include "bart_hasher.h"
+ *  #include <array>
+ *  #include <iostream>
+ *  #include <string>
+ *  #include <unordered_map>
+ *
+ *  int main()
+ *  {
+ *    std::unordered_map<std::array<int, 2>, std::string, butil::Hasher> map = {
+ *      {{0,0}, "all zeros"},
+ *      {{1,1}, "all ones"}
+ *    };
+ *    std::cout << map[{1,1}] << "\n" << map[{0,0}] << std::endl;
```

Comment:
Good example. Just change map to some other word. It's misleading as there does be a data structure called map in std namespace.